src/elements/content-sidebar/SidebarPanels.js

```diff
@@ -243,51 +298,70 @@ class SidebarPanels extends React.Component<Props, State> {
+
+        const hasCustomPanels = customPanels && customPanels.length > 0;
+
+        // Build eligibility for custom panels
+        const customPanelEligibility = {};
+        if (hasCustomPanels && customPanels) {
+            customPanels.forEach(panel => {
+                const canShowPanel = !panel.isDisabled;
+                if (canShowPanel) {
+                    customPanelEligibility[panel.path] = true;
+                }
+            });
+        }
 
         const panelsEligibility = {
-            [SIDEBAR_VIEW_BOXAI]: canShowBoxAISidebarPanel,
             [SIDEBAR_VIEW_DOCGEN]: hasDocGen,
             [SIDEBAR_VIEW_SKILLS]: hasSkills,
             [SIDEBAR_VIEW_ACTIVITY]: hasActivity,
             [SIDEBAR_VIEW_DETAILS]: hasDetails,
             [SIDEBAR_VIEW_METADATA]: hasMetadata,
+            ...customPanelEligibility,
         };
 
         const showDefaultPanel: boolean = !!(defaultPanel && panelsEligibility[defaultPanel]);
 
-        if (!isOpen || (!hasBoxAI && !hasActivity && !hasDetails && !hasMetadata && !hasSkills && !hasVersions)) {
+        if (
+            !isOpen ||
+            (!hasBoxAI && !hasActivity && !hasDetails && !hasMetadata && !hasSkills && !hasVersions && !hasCustomPanels)
+        ) {
             return null;
         }
 
         return (
             <Switch>
-                {canShowBoxAISidebarPanel && (
-                    <Route
-                        exact
-                        path={`/${SIDEBAR_VIEW_BOXAI}`}
-                        render={() => {
-                            this.handlePanelRender(SIDEBAR_VIEW_BOXAI);
-                            return (
-                                <LoadableBoxAISidebar
-                                    contentName={file.name}
-                                    elementId={elementId}
-                                    fileExtension={file.extension}
-                                    fileID={file.id}
-                                    hasSidebarInitialized={isInitialized}
-                                    ref={this.boxAISidebar}
-                                    startMarkName={MARK_NAME_JS_LOADING_BOXAI}
-                                    cache={this.boxAiSidebarCache}
-                                    setCacheValue={this.setBoxAiSidebarCacheValue}
-                                    {...boxAISidebarProps}
-                                />
-                            );
-                        }}
-                    />
-                )}
+                {hasCustomPanels && customPanels &&
+                    customPanels.map(customPanel => {
+                        const {
+                            id: customPanelId,
+                            path: customPanelPath,
+                            component: CustomPanelComponent,
+                        } = customPanel;
+                        const canShowPanel = !customPanel.isDisabled;
+
+                        return canShowPanel ? (
```

Comment:
there are several nested ternaries in this, i think we can use a guard clause to remove at least one:

```
if (isDisabled) {
    return null;
}
```